src/test/app/Vault_test.cpp

```diff
@@ -2946,64 +2960,69 @@ class Vault_test : public beast::unit_test::suite
-                     Account const& charlie,
-                     auto,
-                     Vault& vault,
-                     PrettyAsset const& asset,
-                     auto&&...) {
-            testcase("IOU non-transferable");
-
-            auto [tx, keylet] = vault.create({.owner = owner, .asset = asset});
-            env(tx);
-            env.close();
+        testCase(
+            [&, this](
+                Env& env,
+                Account const& owner,
+                Account const& issuer,
+                Account const& charlie,
+                auto,
+                Vault& vault,
+                PrettyAsset const& asset,
+                std::function<MPTID(ripple::Keylet)> issuanceId) {
+                testcase("IOU non-transferable");
 
-            tx = vault.deposit(
-                {.depositor = owner, .id = keylet.key, .amount = asset(100)});
-            env(tx);
-            env.close();
+                auto [tx, keylet] =
+                    vault.create({.owner = owner, .asset = asset});
+                tx[sfScale] = 0;
+                env(tx);
+                env.close();
 
-            // Disable DefaultRipple
-            env(fset(issuer, 0, asfDefaultRipple));
-            env.close();
+                {
+                    // Charlie cannot deposit
+                    auto tx = vault.deposit(
+                        {.depositor = charlie,
+                         .id = keylet.key,
+                         .amount = asset(100)});
+                    env(tx, ter{terNO_RIPPLE});
+                    env.close();
+                }
 
-            env(tx, ter{terNO_RIPPLE});
-            env.close();
+                {
+                    PrettyAsset shares = issuanceId(keylet);
+                    auto tx1 = vault.deposit(
+                        {.depositor = owner,
+                         .id = keylet.key,
+                         .amount = asset(100)});
+                    env(tx1);
+                    env.close();
 
-            // Issuer does not need rippling to deposit or withdraw
-            env(vault.deposit(
-                {.depositor = issuer, .id = keylet.key, .amount = asset(100)}));
-            env(vault.withdraw(
-                {.depositor = issuer, .id = keylet.key, .amount = asset(100)}));
-            env.close();
+                    // Charlie cannot receive funds
+                    auto tx2 = vault.withdraw(
+                        {.depositor = owner,
+                         .id = keylet.key,
+                         .amount = shares(100)});
+                    tx2[sfDestination] = charlie.human();
+                    env(tx2, ter{terNO_RIPPLE});
+                    env.close();
 
-            tx = vault.withdraw(
-                {.depositor = owner, .id = keylet.key, .amount = asset(50)});
-            env(tx, ter{terNO_RIPPLE});
-            env.close();
+                    /*
+                    env(pay(owner, charlie, shares(100)));
```

Comment:
@ximinez would you expect this to succeed or to fail, given that ripping on the asset is disabled for Charlie ?